src/resource/solidDataset.test.ts

```diff
@@ -2859,6 +2863,52 @@ describe("createContainerInContainer", () => {
     );
   });
 
+  it("uses the full location URL if absolute thereby applying server-side normalisation", async () => {
+    const mockFetch = setMockOnFetch(
+      jest.fn(window.fetch),
+      mockResponse("Arbitrary response", {
+        headers: {
+          Location: "https://some.pod/parent-container/child-container/",
+        },
+        url: "https://some.pod/parent-container//",
+      })
+    );
+
+    const savedSolidDataset = await createContainerInContainer(
+      "https://some.pod/parent-container//",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    expect(savedSolidDataset!.internal_resourceInfo.sourceIri).toBe(
+      "https://some.pod/parent-container/child-container/"
+    );
+  });
+
+  it("uses the relative location URL to indicate source IRI", async () => {
+    const mockFetch = setMockOnFetch(
+      jest.fn(window.fetch),
+      mockResponse("Arbitrary response", {
+        headers: {
+          Location: "./x/y/",
+        },
+        url: "https://some.pod/parent-container//a/b",
+      })
+    );
+
+    const savedSolidDataset = await createContainerInContainer(
+      "https://some.pod/parent-container//a/b",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    expect(savedSolidDataset!.internal_resourceInfo.sourceIri).toBe(
+      "https://some.pod/parent-container//a/x/y/"
```

Comment:
isn't the goal here to strip the additional `/` in `//` ?

Reply:
If the location header is relative, we can't presume which URL normalisation happens on the server.